# Produtos
### Realizar uma normalização dos dados presentes no arquivo produtos_raw.csv. 

In [1]:
#importando bibliotecas
import pandas as pd

In [2]:
# carregando o dataset
df_produtos = pd.read_csv("produtos_raw.csv")

### Análise exploratória para identificar possíveis problemas

In [3]:
df_produtos.head(10)

,name,price,code,actual_category
0,Transponder AIS Maré Magnum,R$ 33122.52,1,ELETRONICOS
1,Transponder Furuno Marlin,R$ 13998.15,2,ELETRONICOS
2,Radar Furuno Pulse Leviathan,R$ 9024.19,3,E L E T R Ô N I C O S
3,Rádio AIS Hydro Tidal Zen,R$ 3381.88,4,Eletrunicos
4,Piloto Automático Furuno Storm,R$ 23669.01,5,Eletronicoz
5,Transponder AIS Vector,R$ 11820.21,6,Eletrunicos
6,Radar AIS Zen,R$ 19518.77,7,eLeTrÔnIcOs
7,GPS AIS Zen,R$ 4984.15,8,E L E T R Ô N I C O S
8,Transponder AIS Titan Pulse,R$ 39705.5,9,Eletronicoz
9,Piloto Automático Simrad Titan Flux Magnum,R$ 32033.04,10,eletrônicos


In [4]:
df_produtos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 157 entries, 0 to 156
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   name             157 non-null    object
 1   price            157 non-null    object
 2   code             157 non-null    int64 
 3   actual_category  157 non-null    object
dtypes: int64(1), object(3)
memory usage: 5.0+ KB


In [5]:
df_produtos['code'].unique()

array([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
        14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,  26,
        27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,  39,
        40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,  52,
        53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,  65,
        66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,  78,
        79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,
        92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104,
       105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117,
       118, 119, 120, 145, 121, 122, 123, 124, 125, 126, 127, 128, 129,
       130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142,
       143, 144, 146, 147, 148, 149, 150])

In [6]:
df_produtos['actual_category'].unique()

array(['ELETRONICOS', 'E L E T R Ô N I C O S', 'Eletrunicos',
       'Eletronicoz', 'eLeTrÔnIcOs', 'eletrônicos', 'Eletrônicos',
       'Eletroniscos', 'Eletronicos', 'eletronicos', 'EletrônicoS',
       'ELEtRÔNICOS', 'PROPULSAO', 'Propulção', 'Prop', 'Propulssão',
       'propulsao', 'P R O P U L S Ã O', 'Propução', 'propulsão',
       'pRoPuLsÃo', 'Propulçao', 'Propulsam', 'PrOpUlSãO', 'Ancoragem',
       'AnCoRaGeM', 'Encoragem', 'Ancoraguem', 'Ancorajm', 'AncorageM',
       'A N C O R A G E M', 'ANCORAGEM', 'aNcOrAgEm', 'Ancorajem',
       'Encoragi', 'ancoragem', 'Ancorajen', 'AncorajeM', 'Ancoragen'],
      dtype=object)

### Pontos mais chamativos após análise exploratória
- Temos 150 codigos únicos, mas 157 linhas, indicando duplicatas
- Preço está em tipo string
- Categorias com grafia não padronizada/errada, dificultando agregação.

### Parte 1 - Padronização de categorias

In [7]:
def padronizar_categoria(txt):
    txt = str(txt).upper().replace(" ", "") # Remove espaços e coloca em caixa alta
    if 'ELETR' in txt:
        return 'eletrônicos'
    elif 'PROP' in txt:
        return 'propulsão'
    elif 'ANCOR' in txt or 'ENCOR' in txt:
        return 'ancoragem'
    return 'outros'

df_produtos['actual_category'] = df_produtos['actual_category'].apply(padronizar_categoria)

In [8]:
df_produtos['actual_category'].unique()

array(['eletrônicos', 'propulsão', 'ancoragem'], dtype=object)

### Parte 2 - Conversão dos valores de preço de tipo string para numérico float

In [9]:
# Remove o prefixo 'R$ ' e converte a coluna price para float
df_produtos['price'] = df_produtos['price'].str.replace('R$ ', '', regex=False).astype(float)

In [10]:
df_produtos['price'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 157 entries, 0 to 156
Series name: price
Non-Null Count  Dtype  
--------------  -----  
157 non-null    float64
dtypes: float64(1)
memory usage: 1.4 KB


### Parte 3 - Remoção de Duplicatas

In [11]:
# existem mais linhas que valores únicos em id
df_produtos = df_produtos.drop_duplicates()

In [12]:
# Visualizando o resultado final
print("Dados normalizados com sucesso.")
df_produtos.head()

Dados normalizados com sucesso.


,name,price,code,actual_category
0,Transponder AIS Maré Magnum,33122.52,1,eletrônicos
1,Transponder Furuno Marlin,13998.15,2,eletrônicos
2,Radar Furuno Pulse Leviathan,9024.19,3,eletrônicos
3,Rádio AIS Hydro Tidal Zen,3381.88,4,eletrônicos
4,Piloto Automático Furuno Storm,23669.01,5,eletrônicos


In [13]:
df_produtos.info()

<class 'pandas.core.frame.DataFrame'>
Index: 150 entries, 0 to 156
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             150 non-null    object 
 1   price            150 non-null    float64
 2   code             150 non-null    int64  
 3   actual_category  150 non-null    object 
dtypes: float64(1), int64(1), object(2)
memory usage: 5.9+ KB
